bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/AnnotationUtils.java

```diff
@@ -109,7 +109,7 @@ private static void processObjectMethodLambdaAnnotation(MapValue globalAnnotMap,
     public static void processFPValueAnnotations(FPValue fpValue, MapValue globalAnnotMap, String name) {
         BAnnotatableType type = (BAnnotatableType) fpValue.getType();
         BString nameKey = StringUtils.fromString(name);
-        if (globalAnnotMap.containsKey(nameKey)) {
+        if (globalAnnotMap != null && globalAnnotMap.containsKey(nameKey)) {
```

Comment:
How does this become null now?

Reply:
This issue is already there in the master. Even for the following, I get the NPE.
```
function func() returns int {
    return 1;
}
boolean b = func is function () returns int;
panic error(string `${b}`);
```

Reply:
```ballerina
function func() returns int {
    return 1;
}
public function main() {
    boolean b = func is function () returns int;
    panic error(string `${b}`);
}
```
seems to work for me.

Reply:
This doesn't work for me. Not sure why. I have reverted this change anyway. Will check whether the checks pass without this change

Reply:
The build fails with an NPE without this change. Hence added back the null check.

Reply:
Doesn't that mean this is becoming null due to a change in this PR? Shall we try and figure out why?

Reply:
Since this happens only when writing unit tests, need to properly investigate the cause for this. Created https://github.com/ballerina-platform/ballerina-lang/issues/30017 to track this. To avoid the null check for annotation map, a cast was added for the runtime tests to avoid expressions that always evaluate to true.